apps/sharedtest/test/unit/wifi_helper_test.js

```diff
@@ -5,6 +5,157 @@ require('/shared/js/wifi_helper.js');
+      password = 'ABCDEF123456ABCDEF123456ABABCDEF123456ABCDEF123456AB123456';
+      assert.isTrue(WifiHelper.isValidInput(network_security, password),
+        'Input should be valid with 58 HEX char');
+
+      // 58 HEX invalid char
+      password = 'ABCGH12345ABCGH12345123456ABCGH12345ABCGH12345123456ABCGH1';
+      assert.isFalse(WifiHelper.isValidInput(network_security, password),
+        'Input should be invalid with 58 not HEX char');
+    });
+
+    test('Test WPA-PSK network >', function() {
+      var network_security = 'WPA-PSK';
+      var password = '';
+
+      assert.isFalse(WifiHelper.isValidInput(network_security, password),
+        'Input should be invalid if not using password');
+
+      password = '12345';
+      assert.isFalse(WifiHelper.isValidInput(network_security, password),
+        'Input should be invalid if password is too short');
+
+      password = '1234567';
+      assert.isFalse(WifiHelper.isValidInput(network_security, password),
+        'Input should be invalid if password is too short');
+
+      password = '12345678';
+      assert.isTrue(WifiHelper.isValidInput(network_security, password),
+        'Input should be valid if password is long enough');
+
+      password = 'QWERTY123456';
+      assert.isTrue(WifiHelper.isValidInput(network_security, password),
+        'Input should be valid if password is long enough');
+    });
+
+    suite('> WPA-EAP network', function() {
+      var password;
+      var identity;
+      var network_security;
+      var eap;
+
+      setup(function() {
+        password = '';
+        identity = '';
+        network_security = '';
+        eap = '';
+      });
+
+      test('Any input is valid with SIM >', function() {
+        network_security = 'WPA-EAP';
+        password = '';
+        identity = '';
+        eap = 'SIM';
+
+        assert.isTrue(WifiHelper.isValidInput(network_security, password,
+          identity, eap), 'Any input should be valid with a SIM network');
+      });
+
+      test('if password and identify are empty', function() {
+        network_security = 'WPA-EAP';
+        password = '';
+        identity = '';
+
+        assert.isFalse(WifiHelper.isValidInput(network_security, password, 
+          identity), 
+          'Input should be invalid when password and identity not filled');
+      });
+
+      test('if we have password, but no identify', function() {
+        network_security = 'WPA-EAP';
+        password = '123'; // valid password
+        identity = '';
+
+        assert.isFalse(WifiHelper.isValidInput(network_security, password,
+          identity), 
+          'Input should be invalid when password and identity not filled');
+      });
+
+      test('if we have password and identify', function() {
+        network_security = 'WPA-EAP';
+        password = '123'; // valid password
+        identity = 'identification'; // valid identity 
+
+        assert.isTrue(WifiHelper.isValidInput(network_security, password,
```

Comment:
either the remark or assertion is wrong here? You assert isValidInput returns true, but say Input should be invalid.